tools/ExampleExtractor/ExtractorAndTesterUsersGuide.md

```diff
@@ -0,0 +1,483 @@
+> ```
+````
+
+## Other Information
+
+### Unsafe Code
+
+All templates support compilation of examples containing unsafe code.
+
+### Examples Containing Pseudo-Code
+
+Some examples contain C# grammar-rule names, which show a general format rather than source code that will compile. Such names are fenced using the following notation: `«rule_name»`. These examples do *not* have *example_annotation*s.
+
+Here’s an example:
+
+> When a *resource_acquisition* takes the form of a *local_variable_declaration*, it is possible to acquire multiple resources of a given type. A `using` statement of the form
+>
+> ```csharp
+> using (ResourceType r1 = e1, r2 = e2, ..., rN = eN) «statement»
+> ```
+>
+> is precisely equivalent to a sequence of nested `using` statements:
+>
+> ```csharp
+> using (ResourceType r1 = e1)
+> using (ResourceType r2 = e2)
+> ...
+> using (ResourceType rN = eN)
+> «statement»
+> ```
+
+### Using Chevron-Quotes for Emphasis
+
+In the Expressions chapter, we use «…» notation for emphasis, as follows:
+
+> ```csharp
+> struct Color
+> {
+>     public static readonly Color White = new Color(...);
+>     public static readonly Color Black = new Color(...);
+>     public Color Complement() {...}
+> }
+>
+> class A
+> {
+>     public «Color» Color;              // Field Color of type Color
+>
+>     void F()
+>     {
+>         Color = «Color».Black;         // Refers to Color.Black static member
+>         Color = Color.Complement();  // Invokes Complement() on Color field
+>     }
+>
+>     static void G()
+>     {
+>         «Color» c = «Color».White;       // Refers to Color.White static member
+>     }
+> }
+> ```
+>
+> For expository purposes only, within the `A` class, those occurrences of the `Color` identifier that reference the `Color` type are delimited by `«...»`, and those that reference the `Color` field are not.
+
+Clearly, this code won’t compile, as is. As such, the ExampleExtractor tool removes the delimiters, leaving identifiers, and the code compiles.
+
+### Tips for Creating New Testable Examples
+
+Here are some things to consider:
+
+Always add an *example_annotation* to a new example that is intended to be tested. If the annotations details are not completely known, use something like the following:
+
+````
+<!-- UntestedExample: {template:"x", name:"x", replaceEllipsis:true, expectedOutput:["x", "x"], expectedErrors:["x","x"], expectedWarnings:["x","x"], ignoredWarnings:["x","x"], expectedException:"x"} -->
+````
+
+That way, a person or tool can easily find untested examples, resolve them, and fill-in the missing information.
+
+If the runtime behavior is implementation-defined, and the annotation is incomplete, use 
+
+````
+<!-- ImplementationDefinedExample: { … } -->
+````
+
+If the runtime behavior is undefined, and the annotation is incomplete, use 
+
+````
+<!-- UndefinedExample: { … } -->
+````
+
+If the example involves multiple source files, use
+````
+<!-- RequiresSeparateFilesExample: { … } -->
+````
```

Comment:
It may be worth adding that this grep command can then be used to find all such examples:

```sh
 grep -E '<!-- [A-Za-z]+Example' standard/*.md
```

Reply:
Done